File: bookvillage/front/src/service/customerService.js
import $axiosInst from '@/axios/AxiosInst'

export class customerService {


    loginCustomer(customerObj){
        return $axiosInst
            .post('/customer/login' , customerObj)
            .then( response => {
                return response.data;
            }).catch( error => {
                return error;
            })
    }

    addCustomer(customerObj){
        return $axiosInst
            .post('/customer/add' , customerObj)
            .then( response => {
                return response.data;
            }).catch( error => {
                return error;
            })
    }


    getCustomerList(){
        return $axiosInst
            .get('/customer/list')
            .then( response => {
                return response.data;
            }).catch( error => {
                return error;
            })
    }


    getCustomerDetail(userSeq) {
        return $axiosInst
            .get('/customer/detail/' + userSeq)
            .then(response => {
                return response.data;
            }).catch(error => {
                return error;
            })
    }

    getCustomerDetailById() {
        return $axiosInst
            .post('/customer/detail' )
            .then(response => {
                return response.data;
            }).catch(error => {
                return error;
            })
    }


    updateCustomer(customerObj) {
        return $axiosInst
            .put('/customer/update/' , customerObj)
            .then( response => {
                return response.data;
            }).catch( error => {
                return error;
            })
    }

    deleteCustomer(userSeq){
         return $axiosInst
             .delete('/customer/delete/' + userSeq)
             .then( response => {
                 return response;
             }).catch( error => {
                 return error;
             })
    }



    }

File: bookvillage/src/main/java/com/mi/bookvillage/model/service/CustomerService.java
package com.mi.bookvillage.model.service;

import com.mi.bookvillage.model.dao.CustomerDAO;
import com.mi.bookvillage.model.vo.CustomerVO;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CustomerService {

    private CustomerDAO customerDAO;
    private BCryptPasswordEncoder encoder;

    public CustomerService(CustomerDAO customerDAO , BCryptPasswordEncoder encoder){
        this.customerDAO = customerDAO;
        this.encoder = encoder;
    }

    public List<CustomerVO> getCustomerList(){
        return customerDAO.getCustomerList();
    }


    public CustomerVO loginCustomer(CustomerVO customerVO){
        CustomerVO authCustomer = customerDAO.loginCustomer(customerVO);
        if(authCustomer == null || !encoder.matches( customerVO.getUserPw(), authCustomer.getUserPw())){
            throw new RuntimeException("password can not be null or unmatched");
        }
        return authCustomer;
    }


    public void addCustomer(CustomerVO customerVO){
        String encodedPassword = encoder.encode(customerVO.getUserPw());
        customerVO.setUserPw(encodedPassword);
        customerDAO.addCustomer(customerVO);
    }


    public CustomerVO getCustomerDetailById(String userId){
        CustomerVO customer = customerDAO.getCustomerDetailById(userId);
        if(customer == null){
            throw new RuntimeException("Invalid userInfo");
        }
        return customer;
    }


    public CustomerVO getCustomerDetailBySeq(int userSeq){
        CustomerVO customer = customerDAO.getCustomerDetailBySeq(userSeq);
        if(customer == null){
            throw new RuntimeException("Invalid userInfo");
        }
        return customer;
    }


    public void updateCustomer(CustomerVO customerVO){
        if( customerVO.getUserPw() != null ){
            String encodedPassword = encoder.encode(customerVO.getUserPw());
            customerVO.setUserPw(encodedPassword);
        }
        customerDAO.updateCustomer(customerVO);
    }
    public void deleteCustomer(int userSeq){
        customerDAO.deleteCustomer(userSeq);
    }

}
